src/NUnitFramework/framework/Constraints/AnyOfConstraint.cs

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2007 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// <see cref="AnyOfConstraint"/> is used to determine whether item is equal to any of parameters
+    /// </summary>
+    public class AnyOfConstraint : Constraint
+    {
+        private readonly IEnumerable _expected;
+
+        /// <summary>
+        /// The NUnitEqualityComparer in use for this constraint
+        /// </summary>
+        private readonly NUnitEqualityComparer _comparer = NUnitEqualityComparer.Default;
+
+        /// <summary>
+        /// Construct a OneOfConstraint
+        /// </summary>
+        /// <param name="expected">Expected collection</param>
+        public AnyOfConstraint(IEnumerable expected)
+        {
+            _expected = expected;
+        }
+
+        /// <summary>
+        /// The Description of what this constraint tests, for
+        /// use in messages and in the ConstraintResult.
+        /// </summary>
+        public override string Description
+        {
+            get
+            {
+                return "any of " + MsgUtils.FormatValue(_expected);
+            }
+        }
+
+        /// <summary>
+        /// Test whether item is present in expected collection
+        /// </summary>
+        /// <typeparam name="TActual">Actual item type</typeparam>
+        /// <param name="actual">Actual item</param>
+        /// <returns>A <see cref="ConstraintResult"/> indicating whether item is present 
+        /// in expected collection.</returns>
+        public override ConstraintResult ApplyTo<TActual>(TActual actual)
+        {
+            var tolerance = Tolerance.Default;
```

Comment:
Why not Tolerance.Exact? Since you decided not to support tolerance (Within, etc.) on this constraint, there's no way to specify a tolerance with the exception of using the `DefaultFloatingPointTolerance` attribute on the test. So all matches will end up being exact __except__ for floating point. Is that your intention?

Reply:
Never having figured out how to use the `ref tolerance` parameter, to me this code reads as "I don't care about tolerance; give me what people want by default."

Reply:
Well, I think I simply forgot about tolerance-related methods, so I'm fine adding them.

Reply:
So you are you (and @jnm2) good with

1. Applying any command-line default tolerance to a floating point value, rather than zero?
2. __Not__ having any tolerance allowed for any other Types?

BTW, `Tolerance.Exact` is like the "default default". `Tolerance.Default` is changeble using the command-line as well as the attribute.

I'm OK with this if you guys are...

Reply:
I want this to behave however `EqualTo().Or.EqualTo()` and the collection constraints behave when you don't specify a tolerance.

Reply:
Then of course you want Default. I think it deserves a mention in the docs, because I don't think it's necessarily intuitive in the same way it is for `EqualTo`, particularly since we don't allow changing it.

BTW, if we __did__ want to allow changing, I'd argue that this should be a variant constructor for the `EqualConstraint`, with all the same features, rather than creating a common base.

Reply:
I think we have same behavior in CollectionItemsEqualConstraint - Tolerance.Default used, and no way to change that.
While I'm not sure that behavior that you described is very intuitive, I'd say that having it inconsistent across constraints is even worse.

Reply:
We probably should have called the existing `Tolerance.Default` something else, like `Tolerance.Unspecified`. It seems like there is some reason to go through all the constraints and look at their consistency in this regard. A separate, much bigger issue, of course.

Reply:
`CollectionItemsEqualConstraint` is an abstract class. It's up to the individual constraints to determine if a tolerance makes sense in their context. That said, a quick review indicates to me they don't do a good job of it. As mentioned above, I guess our use of tolerance deserves some kind of overall review that we can't deal with in this issue.

Mainly, I brought this up to ensure that your not including `Within` was intentional. Either decision is OK with me.

Reply:
I haven't added `Within` because I wasn't able to come up with a use case. If you think it's worth it - I'll add

Reply:
Nope, like I said, I just wanted it thought about rather than missed. I'm a big fan of explicit decisions and it's sometimes hard just looking at the code to know how a decision was reached.

Reply:
I wouldn't add that at this point. 
I don't think anyone would really need that (especially considering no one asked for that in CollectionEquivalent constraint, for example).
Besides, full Within functionality (with all those seconds, days, percents etc) messes IntelliSense quite a lot.

As you already said, it's much easier to add functionality if needed than remove if not.